### DISCUSSION TITLE
Faction ID

### DISCUSSION THREAD

**Discussion by veselin-ugrinov** [TrinityCore/TrinityCore#26323] [Q&A] [answered]
Hi guys.
Since I am running a crossfaction server, I want to make all the friendly NPC friendly to both alliance and horde, but aggressive against the enemy mobs in the ICC . Should I create a new faction ID, or use an existing one ? Is this doable ?

**Comment by ghost** [comment]
That sounds tricky, because making changes to their factions could affect how faction reputation is received when completing quests that give specific reputation points within their own set of factions (unless you are planning to make it a requirement that they don't do, or expect to do, any reputation quests). I don't know exactly, but it could be more useful to create a new faction which could (maybe) contain all the factions they normally belong to. I also don't know how to set up a configuration like that, because it will be a highly customized modification.

edit: You could always wait for more replies, just in case someone has got a suggestion for how to begin making such changes. Also, this could be more suitable for the https://github.com/TrinityCore/TrinityCoreCustomChanges repository (it does not have any "Discussion" tab yet, so it would be necessary to create a normal issue ticket in that repo).

**Comment by Aokromes** [comment, marked as answer]
what's needed here is to fix cross faction part of pre-gunship and make the mobs on entrance not agressive to other faction.
https://github.com/TrinityCore/TrinityCore/issues/24540

**Comment by ghost** [comment]
What about using sanctuary settings, forced peace, like in Dalaran & Shattrath?

**Reply by veselin-ugrinov** [reply]
Is that doable, and how ?

**Reply by Aokromes** [reply]
> What about using sanctuary settings, forced peace, like in Dalaran & Shattrath?

na, it needs to be done like on forge of souls, pit of saron and so.

**Reply by ghost** [reply]
Do you mean the way the NPCs are changed to faction 35 - friendly - for all players?

**Reply by Aokromes** [reply]
> Do you mean the way the NPCs are changed to faction 35 - friendly - for all players?

yeah somehow like that, if horde raid, all aliance mobs enemy to every member of raid and all horde mobs friendly to every member on raid. but this needs c++ script.
